runtime/browser/runtime.cc

```diff
@@ -225,13 +222,10 @@ void Runtime::DidNavigateMainFramePostCommit(
 
 content::JavaScriptDialogManager* Runtime::GetJavaScriptDialogManager(
     content::WebContents* web_contents) {
-#if defined(OS_LINUX)
+#if defined(USE_AURA)
   return app_modal::JavaScriptDialogManager::GetInstance();
 #else
-  if (!javascript_dialog_manager_) {
-    javascript_dialog_manager_.reset(new RuntimeJavaScriptDialogManager);
-  }
-  return javascript_dialog_manager_.get();
```

Comment:
Plus if you carefully read the message and open the original https://github.com/crosswalk-project/crosswalk/commit/b19a154ecbb1bd422b0489cac33c7aace9aae691 you'll see that I've added this code. Android is not using this method but is using the RuntimeJavaScriptDialogManager.